benchmark/Streamly/Benchmark/Data/Parser/ParserD.hs

```diff
@@ -64,6 +64,20 @@ benchIOSink value name f =
 -- Parsers
 -------------------------------------------------------------------------------
 
+-- XXX Heap gets exhausted with value, hence (value `div` 2) was used.
+{-# INLINE choice1 #-}
+choice1 :: MonadCatch m => Int -> SerialT m a -> m ()
+choice1 value = IP.parseD (PR.choice [PR.fromFold (FL.take value FL.drain)])
+
+{-# INLINE choice2 #-}
+choice2 :: MonadCatch m => Int -> SerialT m a -> m ()
+choice2 value =
+    IP.parseD
+        (PR.choice
+             [ PR.fromFold (FL.take (value `div` 2) FL.drain)
+             , PR.fromFold (FL.take (value `div` 2) FL.drain)
+             ])
+
```

Comment:
We already have a benchmark for `choice` which uses `asum`, we can rename that to `choice (asum)`. For this benchmark we should use exactly the same test as for `asum` to make a fair comparison. Also, just one benchmark is enough.